eq-author-api/db/datastore/datastore-mongodb.test.js

```diff
@@ -340,6 +379,649 @@ describe("MongoDB Datastore", () => {
+          id: "user-1",
+          email: "user1@example.com",
+          name: "Joe Bloggs",
+          externalId: "user-1",
+          picture: "",
+        });
+
+        await mongoDB.createUser({
+          id: "user-2",
+          email: "user2@example.com",
+          name: "Jane Smith",
+          externalId: "user-2",
+          picture: "",
+        });
+
+        await mongoDB.createUser({
+          id: "test-user",
+          email: "test-user@example.com",
+          name: "Test User",
+          externalId: "test-user",
+          picture: "",
+        });
+
+        await mongoDB.createUser({
+          id: "user-4",
+          email: "null@example.com",
+          name: null, // `name` is null to test questionnaires created by users with null `name` are returned
+          externalId: "user-4",
+          picture: "",
+        });
+
+        await mongoDB.createQuestionnaire(
+          mockQuestionnaire({
+            title: "Test questionnaire 1",
+            ownerId: "user-1",
+            shortTitle: "Alias 1",
+            createdAt: new Date(2021, 2, 5, 5, 0, 0, 0),
+          }),
+          ctx
+        );
+        await mongoDB.createQuestionnaire(
+          mockQuestionnaire({
+            title: "Test questionnaire 2",
+            ownerId: "user-1",
+            createdAt: new Date(2021, 2, 10, 5, 0, 0, 0),
+          }),
+          ctx
+        );
+        await mongoDB.createQuestionnaire(
+          mockQuestionnaire({
+            title: "Test questionnaire 3",
+            ownerId: "user-2",
+            editors: ["user-1"],
+            createdAt: new Date(2021, 2, 15, 5, 0, 0, 0),
+          }),
+          ctx
+        );
+        await mongoDB.createQuestionnaire(
+          mockQuestionnaire({
+            title: "Test questionnaire 4",
+            ownerId: "user-2",
+            createdAt: new Date(2021, 2, 20, 5, 0, 0, 0),
+          }),
+          ctx
+        );
+        // ** "Test questionnaire 5" is not included in several test assertions as it is not public and `ctx.user` is not owner/editor
+        await mongoDB.createQuestionnaire(
+          mockQuestionnaire({
+            title: "Test questionnaire 5",
+            ownerId: "user-2",
+            createdAt: new Date(2021, 2, 25, 5, 0, 0, 0),
+            isPublic: false,
+          }),
+          ctx
+        );
+        await mongoDB.createQuestionnaire(
+          mockQuestionnaire({
+            title: "Test questionnaire 6",
+            ownerId: "user-1",
+            createdAt: new Date(2021, 2, 30, 5, 0, 0, 0),
+            isPublic: false,
+          }),
+          ctx
+        );
+        await mongoDB.createQuestionnaire(
+          mockQuestionnaire({
+            title: "Test questionnaire 7",
+            ownerId: "user-4",
+            createdAt: new Date(2021, 4, 10, 5, 0, 0, 0),
+          }),
+          ctx
+        );
+      });
+
+      it("Should return questionnaires with title containing the search term", async () => {
```

Comment:
*should return
Issue throughout

Reply:
Based on a technical discussion held today, all tests in this file have been updated to follow the "should" format (lower case) instead of the previous "Should" format - this includes changes to both the new tests and the pre-existing tests in this file.